components/editor/NoteHeader.tsx

```diff
@@ -39,34 +39,52 @@ export default function NoteHeader() {
     (state) => !state.isSidebarOpen && state.openNoteIds?.[0] === currentNote.id
   );
   const isCloseButtonVisible = useStore(
-    (state) => state.openNoteIds?.[0] !== currentNote.id
+    (state) => state.openNoteIds?.[1]
   );
   const note = useStore((state) => state.notes[currentNote.id]);
 
   const onClosePane = useCallback(() => {
     const currentNoteIndex = store
       .getState()
       .openNoteIds.findIndex((openNoteId) => openNoteId === currentNote.id);
+    const stackedNoteIds = queryParamToArray(stackQuery);
 
     if (currentNoteIndex < 0) {
       return;
     }
 
-    // Remove from stacked notes and shallowly route
-    const stackedNoteIds = queryParamToArray(stackQuery);
-    stackedNoteIds.splice(
-      currentNoteIndex - 1, // Stacked notes don't include the main note
-      1
-    );
-
-    router.push(
-      {
-        pathname: router.pathname,
-        query: { ...router.query, stack: stackedNoteIds },
-      },
-      undefined,
-      { shallow: true }
-    );
+    if (currentNoteIndex == 0) {
+      // Changes Current Note to first note in stack 
+      currentNote.id = stackedNoteIds[0];
+      stackedNoteIds.splice(
+        0, 
+        1
+      );// Removes first note in stack from the stack 
+      router.push(
+        {
+          pathname: router.pathname,
+          query: { ...currentNote, stack: stackedNoteIds },
```

Comment:
Spreading `currentNote` here is kind of weird, because it relies on the implementation of `currentNote` not changing from how it currently is (where it's just an `id`). 

I think it would be better to be more explicit and do `id: currentNote.id` instead

Reply:
Yeah, that makes sense. Thanks for that tip! I wasn't very sure about the types used for router.query and currentNote. I saw it just had an id string so I assumed putting currentNote was fine.

Reply:
id: currentNote.id breaks the query, do you know why? currentNote.id doesn't work either. the query needs an object according to visual studio

Reply:
Try `query: { id: currentNote.id, stack: stackedNoteIds }`